### ISSUE TITLE
problems

### ISSUE THREAD

**Issue by alexeygritsenko** [MarioZ/MadMilkman.Ini#3]
1. Error

```
var list = new string[] { "one", "two", "three" };
            var list2 = new List<string>() { "one", "two", "three" };

            section2.Serialize(list2);
```
1. MadMilkman.Ini.Documentation.chm dont work!(

**Comment by MarioZ** [comment]
Hi,

Regarding the first issue, note that a serialization enables you to serialize a custom class object into section's keys. Each object's public property of a supported type is serialized as a key with a name equal to property's name and value equal to property's value.

So you could serialize a following object:

``` csharp
public class Sample
{
    public string[] Array { get; set; }
    public List<string> List { get; set; }
}

// ...

var sample = new Sample();
sample.Array = new string[] { "one", "two", "three" };
sample.List = new List<string>() { "one", "two", "three" };

section2.Serialize(sample);
```

And the result would be:

``` cfg
[Section 2]
Array={one,two,three}
List={one,two,three}
```

I hope this helps.

Regarding the second issue, do you perhaps mean it's blank, completely empty? In case you do then right-click on CHM file, select "Properties" and under the "General" tab click "Unblock".

Does this solve your issues?

Regards,
Mario

**Comment by alexeygritsenko** [comment]
ok it works!
but if sample.List = null
Serialize ok
but
Deserialize error. please fix it

**Comment by MarioZ** [comment]
Hi,

Please try again with [this latest version](https://github.com/MarioZ/MadMilkman.Ini/raw/master/MadMilkman.Ini.zip).
Does this solve your issue?

Also note that in upcoming days I'll add few enhancements, after which I'll make a new release (including new NuGet and Setup release).

Regards,
Mario

**Comment by alexeygritsenko** [comment]
After replacing the manual to the new library does not start) Well I think correctly so fix have nothing wrong. Make your innovations in the next few days and quickly publish on nuget

**Comment by alexeygritsenko** [comment]
hi, new problem(

```
//....
var afaf = new List<Include>();
            afaf.Add(new Include() { isOn = true, name = "game" });
            proj.includes = afaf;
            section2.Serialize(proj);
```

classes

```
    public class Project
    {
        public string nameProject { get; set; }
        public List<string> additionalFiles { get; set; }
        public List<Include> includes { get; set; }
    }

    public class Include
    {
        public string name { get; set; }
        public string path { get; set; }
        public bool isOn { get; set; }
    }
```

**Comment by alexeygritsenko** [comment]
List-T- Serialize dont work(

**Comment by alexeygritsenko** [comment]
Although I just realized why. INI files are not designed to hold such complex collections. That's why you did not realize it, yes?

**Comment by MarioZ** [comment]
Hi,

Yes I'm afraid List&lt;T&gt; is not a supported type, currently supported property types are listed in IniKey.IsSupportedValueType(Type) method's remarks.

Also yes you are right, the reason is that INI files are not equipped with a hierarchy structure.
I was considering something like the following:

``` cfg
[Parent]
List={Child1,Child2,Child3}

[Parent[Child1]]
...
[Parent[Child2]]
...
[Parent[Child3]]
...
```

But for now I'll postpone introducing this. Nevertheless please let me know what you think of it.

Regards,
Mario

**Comment by alexeygritsenko** [comment]
Yes, I think it's an option. Nothing better comes to mind ... But I did not wait and started challenging Serialize very diferent Objects Json.
Now is simple at me to ini file. For complex Json file
